eta/core/utils.py

```diff
@@ -1449,31 +1448,26 @@ def stop(self):
 
         # Revert `sys.stdout`
         sys.stdout = self._orig_stdout
-        for logger, inds in self._stdout_loggers.items():
-            for idx in inds:
-                logger.handlers[idx].stream = self._orig_stdout
+        for handler in self._stdout_handlers:
+            handler.setStream(self._orig_stdout)
 
         return self.stdout
 
-    def _get_stdout_loggers(self):
-        all_loggers = it.chain(
+    def _get_stdout_handlers(self):
+        loggers = it.chain(
             [logging.getLogger()], logging.Logger.manager.loggerDict.values()
         )
 
-        stdout_loggers = {}
-        for logger in all_loggers:
-            inds = [
-                i
-                for i, handler in enumerate(getattr(logger, "handlers", []))
+        handlers = set()
+        for logger in loggers:
+            for handler in getattr(logger, "handlers", []):
                 if (
                     isinstance(handler, logging.StreamHandler)
-                    and getattr(handler, "stream", None) == sys.stdout
-                )
-            ]
-            if inds:
-                stdout_loggers[logger] = inds
```

Comment:
I wasn't able to repro the bug either, but it looks like previously it was setting the logger to an index rather than the handler. It looks correct now.